test/src/unit-capi-array_schema.cc

```diff
@@ -2359,6 +2359,222 @@ TEST_CASE_METHOD(
   remove_temp_dir(local_fs.file_prefix() + local_fs.temp_dir());
 }
 
+TEST_CASE_METHOD(
+    ArraySchemaFx,
+    "C API: Test array schema drop int attribute and add back as string",
+    "[!mayfail][capi][array-schema][attribute-drop][attribute-add]") {
+  // Create array schema
+  tiledb_array_type_t array_type = GENERATE(TILEDB_DENSE, TILEDB_SPARSE);
+  tiledb_array_schema_t* array_schema;
+  int rc = tiledb_array_schema_alloc(ctx_, array_type, &array_schema);
+  REQUIRE(rc == TILEDB_OK);
+  rc = tiledb_array_schema_set_capacity(ctx_, array_schema, CAPACITY);
+  REQUIRE(rc == TILEDB_OK);
+  rc = tiledb_array_schema_set_cell_order(ctx_, array_schema, CELL_ORDER);
+  REQUIRE(rc == TILEDB_OK);
+  rc = tiledb_array_schema_set_tile_order(ctx_, array_schema, TILE_ORDER);
```

Comment:
For sparse we want to allow duplicates so that when we read in the unordered layout, the sparse unordered with duplicates reader is used.